alacritty_terminal/src/term/mod.rs

```diff
@@ -1934,7 +1923,25 @@ impl ansi::Handler for Term {
                     cell.reset(&template);
                 }
             },
-            ansi::ClearMode::Saved => self.grid.clear_history(),
+            ansi::ClearMode::Below => {
+                for cell in &mut self.grid[self.cursor.point.line][self.cursor.point.col..] {
+                    cell.reset(&template);
+                }
+                if self.cursor.point.line < self.grid.num_lines() - 1 {
+                    self.grid
+                        .region_mut((self.cursor.point.line + 1)..)
+                        .each(|cell| cell.reset(&template));
+                }
+            },
+            ansi::ClearMode::All => {
+                // Scroll the viewport into the history.
+                self.grid.clear_viewport(&template);
+                // Reset the now empty viewport.
+                self.grid.region_mut(..).each(|c| c.reset(&template));
```

Comment:
Instead of resetting all lines here, we can just do the following above `clear_viewport`:

```rust
for i in 0..self.lines().0 {
    self.grid[i].occ = self.cols().0;
}
```